scl/loadbalancer/gen-loadbalancer.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh
+#############################################################################
+# Copyright (c) 2017 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+TARGETS=$confgen_targets
+TARGET_PARAMS=$confgen_parameters
+
+target_cnt=$(echo $TARGETS | wc -w)
+target_id=0
+echo 'channel {'
+for target in $TARGETS
```

Comment:
The `MSEC` max value is a limit of the number of the `target`s. In this case the script do not really need to create the extra `channel`s.

Reply:
I do not understand this limit, but if you ommit the channel definition the final flag would stop the msg processing not just for this but for other destination.
what do i miss?

Reply:
If there are `MSEC + n` target, those `n` targets won't be chosen.  Of course it won't cause any issue, but it enlarges the configuration unnecessarily.
It would be a minor *tweak* not even creating them.

Reply:
check the code, it wont happen, unless they configure more then 2^128 servers destination :)
we do the modulo of the # of targets on MSEC and MSEC is a large number, even in 1 sec there are 1000 different value, so it should be ok up to a dozen target for sure.

Reply:
I think @Kokan just wanted to clarify that `MSEC` is not a large integer, it's a number between 0 and 999.
That is the reason why you may not want to generate more than 1000 channel blocks.

Reply:
I don't have any problems with the current implementation, we just wanted to let you know about this. :)

Reply:
I do not think that this would actually scale to more then a dozen (and also does not seem to be realistic) so we should be fine wit this :)

Reply:
Have you tried this out in real life? Or any kind of emulation with high traffic ? It would be nice to see how balanced it gets.

Reply:
Feri used a less efficient version (using regexp) in production, ask him for details.